python-checks/src/main/java/org/sonar/python/checks/cdk/IamPrivilegeEscalationCheck.java

```diff
@@ -86,28 +82,14 @@ public class IamPrivilegeEscalationCheck extends AbstractCdkResourceCheck {
   );
 
   @Override
-  protected void registerFqnConsumer() {
-    checkFqn("aws_cdk.aws_iam.PolicyStatement", (ctx, call) ->
-      checkPolicyStatement(PolicyStatement.build(ctx, call)));
-
-    checkFqn("aws_cdk.aws_iam.PolicyStatement.from_json", (ctx, call) ->
-      CdkIamUtils.getObjectFromJson(ctx, call).ifPresent(json ->
-        checkPolicyStatement(PolicyStatement.build(ctx, json))));
-
-    checkFqn("aws_cdk.aws_iam.PolicyDocument.from_json", (ctx, call) ->
-      CdkIamUtils.getObjectFromJson(ctx, call).ifPresent(json -> CdkIamUtils.getPolicyStatements(ctx, json)
-        .forEach(statement -> checkPolicyStatement(PolicyStatement.build(ctx, statement)))));
-  }
-
-  private static void checkPolicyStatement(PolicyStatement policyStatement) {
+  protected void checkAllowingPolicyStatement(PolicyStatement policyStatement) {
     ExpressionFlow actions = policyStatement.actions();
     ExpressionFlow resources = policyStatement.resources();
 
     if (actions == null
       || resources == null
       || policyStatement.principals() != null
       || policyStatement.conditions() != null
-      || CdkIamUtils.hasNotAllowEffect(policyStatement.effect())
     ) {
```

Comment:
I'm not sure, but probably there should be `!hasAllowEffect()` in if condition (replacement for `CdkIamUtils.hasNotAllowEffect(policyStatement.effect()`). WDYT?

Reply:
It's already been checked by the parent class.